examples/software-device/rs-software-device.cpp

```diff
@@ -0,0 +1,184 @@
+                                RS2_FORMAT_Z16, depth_intrinsics });
+
+    depth_s.add_read_only_option(RS2_OPTION_DEPTH_UNITS, 0.001f);
+
+
+    auto color_stream = color_s.add_video_stream({  RS2_STREAM_COLOR, 0, 1, texture.x,
+                                texture.y, 60, texture.bpp, 
+                                RS2_FORMAT_RGBA8, color_intrinsics });
+
+    dev.create_matcher(RS2_MATCHER_DLR_C);
+    rs2::syncer sync;
+
+    depth_s.start(sync);
+    color_s.start(sync);
+
+    depth_stream.register_extrinsics_to(color_stream, { { 1,0,0,0,1,0,0,0,1 },{ 0,0,0 } });
+
+    //==================================================//
+    //         End of Software-Only Device Declaration  //
+    //==================================================//
+
+    window app(1280, 1500, "RealSense Capture Example");
+    state app_state;
+
+    register_glfw_callbacks(app, app_state);
+
+    rs2::colorizer color_map; // Save colorized depth for preview
+
+    auto last = std::chrono::high_resolution_clock::now();
+
+    float wave_base = 0.f;
+    rs2::pointcloud pc;
+    rs2::points points;
+    int frame_number = 0;
+
+    std::vector<uint8_t> pixels_depth(W * H * BPP_D, 0);
+
+    while (app) // Application still alive?
+    {
+        draw_text(50, 50, "This point-cloud is generated from a synthetic device:");
+        fill_synthetic_depth_data((void*)pixels_depth.data(), W , H , BPP_D, wave_base);
+
+        auto now = std::chrono::high_resolution_clock::now();
+        if (now - last > wave_time)
+        {
+            app_state.yaw -= 1;
+            wave_base += 0.1;
+            last = now;
+        }
+
+        depth_s.on_video_frame({ pixels_depth.data(), // Frame pixels from capture API
+            [](void*) {}, // Custom deleter (if required)
+            W*BPP_D, BPP_D, // Stride and Bytes-per-pixel
+            (rs2_time_t)frame_number * 16, RS2_TIMESTAMP_DOMAIN_HARDWARE_CLOCK, frame_number, // Timestamp, Frame# for potential sync services
+            depth_stream });
+
+
+        color_s.on_video_frame({ texture.frame.data(), // Frame pixels from capture API
+            [](void*) {}, // Custom deleter (if required)
+            texture.x*texture.bpp, texture.bpp, // Stride and Bytes-per-pixel
+            (rs2_time_t)frame_number * 16, RS2_TIMESTAMP_DOMAIN_HARDWARE_CLOCK, frame_number, // Timestamp, Frame# for potential sync services
+            color_stream });
+
+        ++frame_number;
+
+        rs2::frameset fset = sync.wait_for_frames();
+        rs2::frame depth = fset.first_or_default(RS2_STREAM_DEPTH);
+        rs2::frame color = fset.first_or_default(RS2_STREAM_COLOR);
+
+        if (depth && color)
+        {
+            if (auto as_depth = depth.as<rs2::depth_frame>())
+                points = pc.calculate(as_depth);
+            pc.map_to(color);
+
+            // Upload the color frame to OpenGL
+            app_state.tex.upload(color);
+        }
+        draw_pointcloud(app, app_state, points);
+    }
+
```

Comment:
I would split the logic in this function more strictly to: synthetic input creation, SW device creation, app logic (pointcloud / colorizer / rendering). the point is to make the SW device manipulations stand out, and now it is not clear what is the minimum I must take for my own app.